Slim/Control/Queries.pm

```diff
@@ -4734,25 +4740,36 @@ sub worksQuery {
 		$sql .= ' ';
 	}
 
-	$sql .= " GROUP BY $groupBy ";
-
 	my $dbh = Slim::Schema->dbh;
 
 	# Get count of unique composers, the count is cached until the next rescan done event
-	my $cacheKey = md5_hex($sql . join( '', @{$p} ) . 'composerCount' . Slim::Music::VirtualLibraries->getLibraryIdForClient($client));
+	my $ccSql = sprintf($sql, "COUNT(DISTINCT composer.id)");
+	my $cacheKey = md5_hex($ccSql . join( '', @{$p} ) . Slim::Music::VirtualLibraries->getLibraryIdForClient($client));
 
 	my $composerCount = $cache->{$cacheKey};
 	if ( !$composerCount ) {
-		my $ccSql = sprintf($sql, "composer.id AS cid");
-		my $total_sth = $dbh->prepare_cached( qq{
-			SELECT COUNT(DISTINCT cid)  FROM ( $ccSql ) AS t1
-		} );
+		my $total_sth = $dbh->prepare_cached($ccSql);
 
 		$total_sth->execute( @{$p} );
 		($composerCount) = $total_sth->fetchrow_array();
 		$total_sth->finish;
 	}
 
+	# Get count of all results, the count is cached until the next rescan done event
+	my $cSql = sprintf($sql, "COUNT(DISTINCT works.id||composer.id)");
```

Comment:
Concatenating IDs wouldn't work reliably: `1` and `21` would result in the same `121` as `12` and `1`.

Reply:
Good point. I'll amend it.

Reply:
Reverted to GROUP BY for the counts, but just selecting a dummy column value: turns out to be quicker since the previous GROUP BY change for the full query.

Pushes done.